impala/dbapi/interface.py

```diff
@@ -167,4 +229,17 @@ def _bind_parameters(operation, parameters):
             string_parameters[name] = "'" + _escape(value) + "'"
         else:
             string_parameters[name] = str(value)
+
+    # paramstyle is pyformat, but support named as well
+    operation = _replace_named_markers(operation, string_parameters)
+    
+    # but default to pyformat
     return operation % string_parameters
+
+def _bind_parameters(operation, parameters):
+    # If parameters is a list, assume either qmark or numeric
+    # format. If not, assume either named or pyformat parameters
+    if isinstance(parameters, (list, tuple)):
+        return _bind_parameters_list(operation, parameters)
+    else:
```

Comment:
Put another way, should we test `elif isinstance(parameters, dict)` and then raise an exception on the `else` branch?